frontend/pages/docs.vue

```diff
@@ -0,0 +1,45 @@
+<script setup lang="ts">
+const router = useRouter()
+
+const onTocClick = (id: string) => {
+  const el = document.getElementById(id)
+  if (el) {
+    router.push({ hash: `#${id}` })
+    el.scrollIntoView()
+  }
+}
+
+const { data: blogPost } = await useAsyncData('blogToc', () =>
+  queryContent('/documentation').findOne()
+)
+
+const tocLinks = computed(() => blogPost.value?.body.toc.links ?? [])
+
+</script>
+<template>
+  <div class="markdown-body w-full">
+    <ContentDoc path="/documentation">
+      <template #default="{doc}">
+        <div class="flex flex-row w-full">
+          <div class="flex flex-col w-1/4 mx-4 hidden lg:block h-full sticky top-0">
+            <h3> Table of contents </h3>
+            <div class="overflow-y-scroll border rounded-md TableOfContents">
+              <div class="items-center">
+                <DocTableOfContents :toc-links="tocLinks" @click="onTocClick" />
+              </div>
+            </div>
+          </div>
+          <div class="w-3/4 sm:max-lg:w-full p-4">
+            <ContentRenderer :value="doc" class="border-black" />
+          </div>
+        </div>
+      </template>
+    </ContentDoc>
+  </div>
+</template>
```

Comment:
this seems suboptimal. Tried using naive ui's layout component, but it breaks `#` links.

Reply:
What exactly do you find suboptimal?

Reply:
i would imagine that there's supposed to be a way to do the same with components instead of pure html. 

This one seems cluttered. On top of this while playing around with tailwind i've encountered some cases where the css classes break the hash url access as well. 

If the current state of html only seems fine to you, i could roll with it as well.